starter-core/src/test/groovy/io/micronaut/starter/feature/test/HamcrestSpec.groovy

```diff
@@ -0,0 +1,86 @@
+package io.micronaut.starter.feature.test
+
+import io.micronaut.starter.BeanContextSpec
+import io.micronaut.starter.application.ApplicationType
+import io.micronaut.starter.feature.Category
+import io.micronaut.starter.fixture.CommandOutputFixture
+import io.micronaut.starter.options.Language
+import io.micronaut.starter.options.TestFramework
+import io.micronaut.starter.feature.build.gradle.templates.buildGradle
+import io.micronaut.starter.feature.build.maven.templates.pom
+import spock.lang.Shared
+import spock.lang.Subject
+import spock.lang.Unroll
+
+class HamcrestSpec  extends BeanContextSpec implements CommandOutputFixture {
+    @Shared
+    @Subject
+    Hamcrest hamcrest = beanContext.getBean(Hamcrest)
+
+    void 'test readme.md with feature hamcrest contains links to 3rd party docs'() {
+        when:
+        def output = generate(['hamcrest'])
+        def readme = output["README.md"]
+
+        then:
+        readme
+        readme.contains("http://hamcrest.org/JavaHamcrest/")
+    }
+
+    void "test hamcrest belongs to Dev Tools category"() {
+        expect:
+        Category.DEV_TOOLS == hamcrest.category
+    }
+
+    @Unroll
+    void 'test gradle hamcrest feature for language=#language'() {
+        when:
+        String template = buildGradle.template(ApplicationType.DEFAULT, buildProject(), getFeatures(['hamcrest'],
+                language, TestFramework.JUNIT), false).render().toString()
+
+        then:
+        template.contains('testImplementation("org.hamcrest:hamcrest")')
+
+        where:
+        language << Language.values().toList()
+    }
+
+    @Unroll
+    void 'test gradle hamcrest feature fails for language=#language when test framework is not Junit'() {
```

Comment:
Is it not possible to test it with Maven too?

Reply:
yes it is!. done